src/utilities/suggestionList.js

```diff
@@ -0,0 +1,78 @@
+/* @flow */
+/**
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/**
+ * Given a JavaScript value and a GraphQL type, determine if the value will be
+ * accepted for that type. This is primarily useful for validating the
+ * runtime values of query variables.
+ */
+export function suggestionList(
+    input: string,
+    options: Array<string>
+  ): Array<string> {
+  let i;
+  const d = {};
+  const oLength = options.length;
+  for (i = 0; i < oLength; i++) {
+    d[options[i]] = lexicalDistance(input, options[i]);
+  }
+  const result = options.slice();
+  return result.sort((a , b) => d[a] - d[b]);
+}
+
+/**
+ * Computes the lexical distance between strings A and B.
+ *
+ * The "distance" between two strings is given by counting the minimum number
+ * of edits needed to transform string A into string B. An edit can be an
+ * insertion, deletion, or substitution of a single character, or a swap of two
+ * adjacent characters.
+ *
+ * This distance can be useful for detecting typos in input or sorting
+ *
+ * @param {string} a
+ * @param {string} b
+ * @return {int} distance in number of edits
+ */
+function lexicalDistance(a, b) {
```

Comment:
I've seen this couple of times when someone introduces a pretty long field name or mutation response type. They either flip two of the middle words or missed something like the word "Payload" completely. Because at a glance nothing looked misspelled, they thought it was more of a problem with updating the schema and ended up spending a lot of time trying to debug and rerun that.
I actually started with trying to split up the `input` into words and trying to match up `n-1` word, but I didn't think the extra complexity to get that number is going to be significantly better than just `lexicalDistance`. That's why I decided to keep it simple. We can always add the variation for that case when we have evidence that the `lexicalDistance` isn't good enough.